pyadjoint/drivers.py

```diff
@@ -91,3 +92,16 @@ def solve_adjoint(J, tape=None, adj_value=1.0):
 
     with stop_annotating():
         tape.evaluate_adj(markings=False)
+
+
+class marked_controls:
```

Comment:
Document class.
For instance,
"""
Context manager for an iterable of :class:`Control`.
At entry point the controls will all be marked (Made relevant for tape optimizations).
At exit of the context manager, the controls will be reset.
"""

To simplify it, we could just use `contextlib`: https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager